src/main/java/io/seqera/migtool/strategy/DialectQueryBuilder.java

```diff
@@ -0,0 +1,48 @@
+package io.seqera.migtool.strategy;
+
+/**
+ * Build the queries to execute, based on the dialect chosen.
+ * This is needed because Postgres does not support backticks for column names (which is a non-standard adopted by
+ * MySql), so we need to build the queries accordingly.
+ * Since the other dialects supported allow the use of backticks, only two builders are added
+ * {@link PostgresQueryBuilder} and {@link NotPostgresQueryBuilder}.
+ **/
+public abstract class DialectQueryBuilder {
```

Comment:
I found this solution overkill. It would be enough to have a driver pattern having a default implementation (a couple of methods) covering the current statement and the version for Postgres.

Reply:
That kind of things that's simpler to implement, that to describe. If we want to adhere to GoF definition, it would be a [template pattern](https://www.digitalocean.com/community/tutorials/template-method-design-pattern-in-java), but also Jdbc driver using a similar approach, this is why I had used that term. 

Compare [this](https://github.com/seqeralabs/migtool/pull/49) implementation with the one in this PR (I hadn't time to add tests)

Reply:
I considered this approach, but I don't like it because the common SQL needs to be duplicated in both implementations. Meaning that if we need to change one of the queries in one place, we need to change it in both files (Default and Postgres).

In the end, we also needed to add 3 classes here, so I don't see what we gained using this approach. :)

I am proposing a solution now that solves both issues: the one regarding being overkill (since we only have 2 implementations) and the one that implies having the sql string repeated in the subclasses.

Would this be acceptable in your opinion? I think it merges the best of both worlds @pditommaso .